src/main/kotlin/blackjack/model/Participant.kt

```diff
@@ -13,11 +15,18 @@ abstract class Participant {
 
     fun isBusted(): Boolean = (handCards.total > Rules.BLACKJACK_TARGET)
 
-    fun cardsToString(): String {
-        return handCards.cards.joinToString()
-    }
+    fun settlePlayerAndDealer(dealer: Dealer) {
+        earning = when {
+            isBusted()                                                  -> -bet
+            dealer.isBusted()                                           -> +bet
+            handCards.isBlackjack() && dealer.handCards.isBlackjack()   -> 0
+            handCards.isBlackjack()                                     -> (bet * 1.5).toInt()
+            dealer.handCards.isBlackjack()                              -> -bet
+            handCards.total > dealer.handCards.total                    -> +bet
+            handCards.total < dealer.handCards.total                    -> -bet
+            else                                                        -> 0
+        }
```

Comment:
Wow, I've never thought of using when like this; it looks pretty good.
A good way to make it easier to understand quickly might be adding some empty lines to group similar ones :  
```suggestion
    fun settlePlayerAndDealer(dealer: Dealer) {
        earning = when {
            isBusted()                                                  -> -bet
            dealer.isBusted()                                           -> +bet
            
            handCards.isBlackjack() && dealer.handCards.isBlackjack()   -> 0
            handCards.isBlackjack()                                     -> (bet * 1.5).toInt()
            dealer.handCards.isBlackjack()                              -> -bet
            
            handCards.total > dealer.handCards.total                    -> +bet
            handCards.total < dealer.handCards.total                    -> -bet
            
            else                                                        -> 0
        }
```

But this is just a small suggestion, so please do as you please 😉

Reply:
Yes, I agree. Grouping the cases makes it more readable. 😄